src/components/measureGroups/MeasureGroups.tsx

```diff
@@ -702,127 +720,170 @@ const MeasureGroups = () => {
                 />
               )}
               {activeTab === "stratification" && (
-                <div>
-                  {formik.values.stratifications.map((strat, i) => (
-                    <Row>
-                      <Col>
-                        <FieldLabel htmlFor="stratification-select">
-                          Stratification {i + 1}
-                        </FieldLabel>
-                        <TextField
-                          select
-                          id="stratification-select"
-                          label=""
-                          value={formik.values.stratifications[i].cqlDefinition}
-                          inputProps={{
-                            "data-testid": "stratification-select",
-                          }}
-                          onChange={formik.handleChange}
-                          InputLabelProps={{ shrink: false }}
-                          SelectProps={{
-                            native: true,
-                          }}
-                          name={`stratifications[${i}].cqlDefinition`}
-                        >
-                          <option
-                            value=""
-                            data-testid="stratification-select-option"
-                          >
-                            -
-                          </option>
-                          {Object.values(
-                            expressionDefinitions.sort((a, b) =>
-                              a.name.localeCompare(b.name)
+                <FieldArray
+                  name="stratifications"
+                  render={(arrayHelpers) => (
+                    <div>
+                      {formik.values.stratifications ? (
+                        formik.values.stratifications.map(
+                          (strat, i) =>
+                            formik.values.stratifications[i].association !==
+                              "delete" && (
+                              <div key={i}>
+                                <Row>
+                                  <Col>
+                                    <FieldLabel htmlFor="stratification-select">
+                                      Stratification {i + 1} {}
+                                      {formik.values.stratifications.length >
+                                        2 &&
+                                        visibleStrats > 2 && (
+                                          <span>
+                                            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
+                                            <Button
```

Comment:
Can we not use buttons from madie-components, and use madie-design-systems instead ?

Reply:
to come later

Reply:
Done!